### PR TITLE
docs: improve doc comments

### PR DESCRIPTION
**Pull request by rootulp**
Spun out of https://github.com/celestiaorg/celestia-app/pull/691 because these improvements are can be reviewed & merged independently

### CODE REVIEW

File: pkg/shares/shares.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func (n NamespacedShare) Data() []byte {
 }
 
 // NamespacedShares is just a list of NamespacedShare elements.
-// It can be used to extract the raw raw shares.
+// It can be used to extract the raw shares.
 type NamespacedShares []NamespacedShare
```
Comment (evan-forbes) [inline pkg/shares/shares.go:29]: tbh, we should get rid of this struct as its pointless middleware. This is remnant of the first implementation done two years ago

File: x/payment/types/payfordata_test.go [hunk 2]
```diff
@@ -151,7 +151,7 @@ func TestPowerOf2(t *testing.T) {
 	}
 }
 
-// TestCreateCommit only shows if something changed, it doesn't actually show
+// TestCreateCommitment only shows if something changed, it doesn't actually show
```
Comment (evan-forbes) [inline x/payment/types/payfordata_test.go:154]: [optional] while we're editing might as well fix the typo. might need to wrap this line
```suggestion
// TestCreateCommitment only shows if something changed, it doesn't actually show that
```

**Review by evan-forbes — APPROVED** [review]
disabled auto merge to allow the chance to respond to optional change, hope that's okay

File: pkg/shares/shares.go (hunk 1 above)
Comment (rootulp) [reply pkg/shares/shares.go:29]: Sounds good! I can tackle after we merge this PR and https://github.com/celestiaorg/celestia-app/pull/691

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**